stdlib/messaging/kafka/src/main/ballerina/src/kafka/consumer.bal

```diff
@@ -115,27 +119,46 @@ public type ConsumerConfig record {|
 # + timestamp - Timestamp of the record, in milliseconds since epoch.
 # + topic - Topic to which the record belongs to.
 public type ConsumerRecord record {|
-    byte[] key;
-    byte[] value;
+    Data key;
+    Data value;
     int offset;
     int partition;
     int timestamp;
     string topic;
 |};
 
+# In-built Kafka byte array deserializer.
+public const DES_BYTE_ARRAY = "BYTE_ARRAY";
+
+# In-built Kafka string deserializer.
+public const DES_STRING = "STRING";
+
+# In-built Kafka int deserializer.
+public const DES_INT = "INT";
+
+# In-built Kafka float deserializer.
+public const DES_FLOAT = "FLOAT";
+
+# Kafka in-built deserializer type.
+public type DeserializerType DES_BYTE_ARRAY|DES_STRING|DES_INT|DES_FLOAT;
+
 # Represent a Kafka consumer endpoint.
 #
 # + consumerConfig - Used to store configurations related to a Kafka connection.
 public type Consumer client object {
-    *lang:Listener;
+    *'object:Listener;
 
     public ConsumerConfig? consumerConfig = ();
+    private string keyDeserializer = "";
```

Comment:
Since we are setting this value in the constructor, Do we need to initialize this with an empty string?

Reply:
Ah yes. Agreed.